src/profile/components/serviceConnections/ServiceConnection.tsx

```diff
@@ -0,0 +1,180 @@
+import encodeServiceName from '../../helpers/encodeServiceName';
+
+export const STATUS_NONE = 0;
+export const STATUS_PENDING_CONFIRMATION = 1;
+export const STATUS_LOADING = 2;
+export const STATUS_ERROR = 3;
+export const STATUS_DELETE_FORBIDDEN = 4;
+export const STATUS_DONE = 5;
+export const STATUS_ACKNOWLEDGED = 6;
+export const STATUS_CLOSED = 7;
+
+export type DeletionStatus =
+  | typeof STATUS_NONE
+  | typeof STATUS_PENDING_CONFIRMATION
+  | typeof STATUS_LOADING
+  | typeof STATUS_ERROR
+  | typeof STATUS_DELETE_FORBIDDEN
+  | typeof STATUS_DONE
+  | typeof STATUS_CLOSED
+  | typeof STATUS_ACKNOWLEDGED;
+
+function ServiceConnection(props: {
+  service: ServiceConnectionData;
+  onDeletion: (deletedService: ServiceConnectionData) => void;
+}): React.ReactElement | null {
+  const { service, onDeletion } = props;
+
+  const [deletionStatus, setDeletionStatus] = useState<DeletionStatus>(
+    STATUS_NONE
+  );
+
+  const cannotDelete = deletionStatus === STATUS_DELETE_FORBIDDEN;
+  const hasError = deletionStatus === STATUS_ERROR || cannotDelete;
+  const isLoading = deletionStatus === STATUS_LOADING;
+  const isDone = deletionStatus === STATUS_DONE;
+
+  const { t } = useTranslation();
+
+  const getDateTime = (date: Date) => {
+    const day = format(new Date(date), 'dd.MM.yyyy');
+    const time = format(new Date(date), 'HH:mm');
+    return `${day}, ${t('serviceConnections.clock')} ${time}`;
+  };
+
+  const [deleteConnection, authorizationCodeState] = useDeleteServiceConnection(
+    service.name,
+    {
+      fetchPolicy: 'no-cache',
+      onError: errorEvent => {
+        if (errorEvent.graphQLErrors) {
+          errorEvent.graphQLErrors.forEach(graphQlError => {
+            Sentry.captureException(new Error(graphQlError.message));
+          });
+        } else {
+          Sentry.captureException(errorEvent);
+        }
+        setDeletionStatus(STATUS_ERROR);
+      },
+      onCompleted: data => {
+        if (!data.deleteMyServiceData?.result?.success) {
+          setDeletionStatus(STATUS_DELETE_FORBIDDEN);
+        } else {
+          setDeletionStatus(STATUS_DONE);
+        }
+      },
+    }
+  );
+
+  const getModalStatus = () => {
+    const authorizationIsActive =
+      authorizationCodeState.isLoading ||
+      authorizationCodeState.hasCode ||
+      authorizationCodeState.isDeleting;
+
+    if (deletionStatus === STATUS_NONE && authorizationIsActive) {
+      return STATUS_LOADING;
+    }
+
+    return deletionStatus;
+  };
+
+  const onDeleteClick = () => {
+    if (isLoading) {
+      return;
+    }
+    setDeletionStatus(STATUS_PENDING_CONFIRMATION);
+  };
+
+  const onClose = () => {
+    if (isLoading) {
+      return;
+    }
+    if (isDone) {
+      setDeletionStatus(STATUS_ACKNOWLEDGED);
+      onDeletion(service);
+      return;
+    }
+    if (hasError) {
+      setDeletionStatus(STATUS_CLOSED);
+      return;
+    }
+    setDeletionStatus(STATUS_CLOSED);
```

Comment:
The `if (hasError)` block seems to do exactly the same as the rest of the function. Can it be removed?